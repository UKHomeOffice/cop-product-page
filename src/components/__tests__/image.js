import React from "react";
import renderer from "react-test-renderer";
import Image from "../image";

describe("image", () => {
  const link = 'www.google.com'
  it("renders correctly", () => {
    const tree = renderer
      .create(<Image/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when not on mobile", () => {
  const tree = renderer
    .create(<Image noMobile={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

  it("Returns the correct source and alt with its correct values given", () => {
    const tree = renderer
      .create(<Image alt={"description"} source={link}/>);
    const testInstance = tree.root;
    expect(testInstance.props).toHaveProperty("alt", "description");
    expect(testInstance.props).toHaveProperty("source", link);
  });

  it("Returns the correct class properties", () => {
    const tree = renderer
      .create(<Image noImage={true} alt={"description"} source={link}/>);
    const testInstance = tree.root;
    expect(testInstance.props).toHaveProperty("noImage", true);
  });
});