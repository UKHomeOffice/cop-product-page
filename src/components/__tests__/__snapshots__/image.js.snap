// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image renders correctly 1`] = `
<img
  style={
    Object {
      "maxWidth": "100%",
    }
  }
/>
`;

exports[`image renders correctly when not on mobile 1`] = `
<img
  className="noMobile-image"
  style={
    Object {
      "maxWidth": "100%",
    }
  }
/>
`;
